### PR TITLE
Bug - libbeat publisher pipeline output module exits if output returns error.

### PR DESCRIPTION
**Pull request by amitshinde001**
Unlike net client worker, the client worker exits if output returns the error.  Ideally it should break the loop, like net client worker if publish of any output type returns error.

### CODE REVIEW

**Comment by amitshinde001** [comment]
May I know how to re-trigger the failed build?

**Comment by jsoriano** [comment]
jenkins, test this please

**Comment by amitshinde001** [comment]
jenkins, test this please

**Comment by amitshinde001** [comment]
jenkins, test this please

**Comment by amitshinde001** [comment]
> 
> 
> jenkins, test this please

**Comment by amitshinde001** [comment]
Wondering if someone can review this?

**Comment by kvch** [comment]
What is the bug you are trying to resolve? What is the issue it caused?

Publisher pipeline exits otherwise the console output fails. This can be changed if needed, but firt I am interested in the problem you are trying to solve.

**Comment by amitshinde001** [comment]
> 
> 
> What is the bug you are trying to resolve? What is the issue it caused?
> 
> Publisher pipeline exits otherwise the console output fails. The procedure can be changed if needed, but first I am interested in the problem you are trying to solve.

First and foremost thing the behavior of clientWorker and netClientWorker is inconsistent. I am writing two different output modules for beats, one is based on clientWorker and other is based on netClientWorker. Returning error from these two different outputs results in two different behaviors, one exits and other continues. There are no clear guidelines for return value, what action will be taken for error and nil values. Shouldn't we align behavior to one common baseline?

**Comment by kvch** [comment]
I see your point. We should provide guidelines for such situations.

I am ok with moving this PR forward. Could you please move the changelog entry to `CHANGELOG-developer.next.asciidoc` and reword it to something like "make the behaviour of clientWorker and netClientWorker consistent"?

**Comment by amitshinde001** [comment]
> 
> 
> I see your point. We should provide guidelines for such situations.
> 
> I am ok with moving this PR forward. Could you please move the changelog entry to `CHANGELOG-developer.next.asciidoc` and reword it to something like "make the behaviour of clientWorker and netClientWorker consistent"?

I have modified the message from changelog files.

Any idea about back-porting to 7.1?

**Comment by kvch** [comment]
We are not planning to release a new patch version of 7.1. Also, this change might break other devs' work, so I would not backport this PR.

**Comment by amitshinde001** [comment]
> WFG

Wondering what is the merging process? Anything else required from my side?
